test/infrastructure/inmemory/internal/cloud/runtime/cache/client.go

```diff
@@ -424,13 +421,18 @@ func (c *cache) doTryDeleteLocked(resourceGroup string, tracker *resourceGroupTr
 		if err := c.beforeUpdate(resourceGroup, oldObj, obj); err != nil {
 			return false, apierrors.NewBadRequest(err.Error())
 		}
+
+		// TODO: (killianmuldoon) Understand if setting this twice is necessary.
```

Comment:
FYI this is necessary because beforeUpdate restores the DeletionTimestamp to empty.

Might be a cleaner solution to call SetDeletionTimestamp only once, after the call to beforeUpdate

Reply:
We were looking into beforeUpdate and it seems impossible that this code is setting the DeletionTimestamp to empty

Reply:
I'd like to take this as a follow-up - will open an "Improvements to watch" issue with this and a few other things from this review if possible.